stimupdated.py

```diff
@@ -0,0 +1,169 @@
+import numpy as np
+import stim
+import tensorcircuit as tc
+
+np.random.seed(0)
+
+tc.set_dtype("complex128")
+
+clifford_one_qubit_gates = ["h", "x", "y", "z", "s"]
+clifford_two_qubit_gates = ["cnot"]
+clifford_gates = clifford_one_qubit_gates + clifford_two_qubit_gates
+
+
+def genpair(num_qubits, count):
+    choice = list(range(num_qubits))
+    for _ in range(count):
+        np.random.shuffle(choice)
+        x, y = choice[:2]
+        yield (x, y)
+
+
+def random_clifford_circuit_with_mid_measurement(num_qubits, depth):
+    c = tc.Circuit(num_qubits)
+    operations = []
+    for _ in range(depth):
+        for j, k in genpair(num_qubits, 2):
+            c.cnot(j, k)
+            operations.append(f"CNOT({j}, {k})")
+        for j in range(num_qubits):
+            gate = np.random.choice(clifford_one_qubit_gates)
+            getattr(c, gate)(j)
+            operations.append(f"{gate.upper()}({j})")
+        measured_qubit = np.random.randint(0, num_qubits)
```

Comment:
I think it gave a different result due to the change in how you use the `randint()` here, and the bug should be coming from other places. (Also, `randint(a, b)` selects random integer from `a <= i <= b`, so it should be `randint(0, num_qubits - 1)` )